src/service-api.ts

```diff
@@ -183,6 +185,92 @@ const plugin: FastifyPluginAsync<GraaspImportZipPluginOptions> = async (fastify,
       return items;
     },
   );
+
+  // download item as zip
+  fastify.get<{ Params: { itemId: string } }>(
+    '/zip-export/:itemId',
+    //{ schema: zipExport },
+    async ({ member, params: { itemId }, log }, reply) => {
+      const addFileToZip = async (item: Item, dirPath) => {
+        // get item and its related data
+        const itemExtra = item.extra as Extra;
+
+        switch (item.type) {
+          case SERVICE_ITEM_TYPE:
+            // TODO: add uploaded files into zip
+            break;
+          case ItemType.DOCUMENT:
+            archive.append(itemExtra.document?.content, { name: `${dirPath}${item.name}.graasp` });
+            break;
+          // TODO: better handle the case of link / app
+          case ItemType.LINK:
+            archive.append(`_source\n${itemExtra.embeddedLink?.url}\nlinkType: 0\n`, {
+              name: `${dirPath}${item.name}.url`,
+            });
+            break;
+          case ItemType.APP:
+            archive.append(`_source\n${itemExtra.app?.url}\nlinkType: 1\n`, {
+              name: `${dirPath}${item.name}.url`,
+            });
+            break;
+          case ItemType.FOLDER:
+            // append description
+            if (dirPath === '') {
+              dirPath = `/${item.name}/`;
+            } else {
+              dirPath = `${dirPath}${item.name}/`;
+            }
+            if (item.description) {
+              archive.append(item.description, { name: `${dirPath}${item.name}.description.html` });
+            }
+            // eslint-disable-next-line no-case-declarations
+            const tasks = iTM.createGetChildrenTaskSequence(member, item.id, true);
+            // eslint-disable-next-line no-case-declarations
+            const subItems = await runner.runSingleSequence(tasks, log);
+            (subItems as Item[]).forEach((subItem) => {
+              addFileToZip(subItem, dirPath);
+            });
+        }
+      };
```

Comment:
As a general comment, you might want to use the `path` package to avoid only dealing with strings. It won't be the case for us, but this package also handles paths for windows server (which would fail with the paths you use currently).